### ISSUE TITLE
Extensibility: API for adding controller actions

### ISSUE THREAD

**Issue by mabolek** [pixelant/pxa_product_manager#208]
Other extensions should be able to add controller actions to existing controllers.

Also: Replace switchableControllerActions, which is deprecated. Should probably be its own issue, but can be done at the same time as this.

Depends on #206

- [x] Implement
- [x] Documentation

**Comment by VladVilya** [comment]
API implemented and documented, but switchableControllerActions not removed
